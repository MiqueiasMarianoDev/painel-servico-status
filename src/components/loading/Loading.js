import React from 'react'
import { Animation } from '../icones/Animation';
import LoadingAnimationData from '../animation/animation_llbrzglt.json'

export const Loading = () => {
  return (
    <div className="loading">
      <div className="spinner"></div>
      <p>Carregando...</p> 
      <Animation animationData={LoadingAnimationData} width={50} height={50}loop={true}/>
    </div>
  );
}