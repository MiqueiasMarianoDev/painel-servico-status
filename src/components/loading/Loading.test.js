import React from 'react';
import { render, screen } from '@testing-library/react';
import { Loading } from './Loading';

describe('Loading', () => {

  test('renders loading message', () => {
    render(<Loading />);
    
    const loadingMessage = screen.getByText(/Carregando.../i);
    expect(loadingMessage).toBeInTheDocument();
  });

  test('renders spinner', () => {
    render(<Loading />);
    
    const spinner = screen.getByRole('progressbar');
    expect(spinner).toBeInTheDocument();
  });

  test('snapshot', () => {
    const {asFragment} = render(<Loading />);
    
    expect(asFragment()).toMatchSnapshot();
  });

});