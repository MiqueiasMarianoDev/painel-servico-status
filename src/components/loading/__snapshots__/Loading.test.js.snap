// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading snapshot 1`] = `
<DocumentFragment>
  <div
    class="loading"
  >
    <div
      class="spinner"
    />
    <p>
      Carregando...
    </p>
  </div>
</DocumentFragment>
`;
